### ISSUE TITLE
`no-useless-undefined` `: allow return undefined in promise.

### ISSUE THREAD

**Issue by christopher-buss** [sindresorhus/eslint-plugin-unicorn#2494]
### Description

In an async function, I want to be able to do return undefined at the end, to denote that I am intentionally not returning a value. Some ESLint Promise rules expect a return statement to silence specific warnings.

This could be added as an option, `checkPromises`.

For example, if I do:
```ts
const promise = somePromise.then(async () => {
	// some work
});
// `Each then() should return a value or throw - eslint-plugin-promise no-return`
```

To silence this warning, I would just do:
```ts
const promise = somePromise.then(async () => {
	// some work
	return undefined;
});
// `Do not use useless `undefined` - unicorn/no-useless-undefined`
```

### Fail

```json
"unicorn/no-useless-undefined": [ "error", { checkPromises: false }]
```

```ts
const promise = somePromise.then(async () => {
	// some work
	return undefined;
});
//
```

### Pass

```json
"unicorn/no-useless-undefined": [ "error", { checkPromises: true }]
```

```ts
const promise = somePromise.then(async () => {
	// some work
	return undefined;
});
//
```

**Comment by christopher-buss** [comment]
It's necessary due to conflicts with other ESLint packages, and I don't want to disable the rule because I think it's useful. But I guess there will be no resolution here.
